lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -146,8 +146,12 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
             // So that data retrieval information for MPU's and
             // regular puts are stored in the same data structure,
             // place the retrieval info here into a single element array
-            const { key, dataStoreName, dataStoreType, dataStoreETag } =
+            const { key, dataStoreName, dataStoreType,
+                    dataStoreETagFromGetInfo } =
```

Comment:
beware, you're changing the API here by expecting a `dataStoreETagFromGetInfo` field in the `dataGetInfo` object, I think this is not what you want here.